From 2abfc41e4473c4956866e15f7699070200233bd6 Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Mon, 11 Aug 2014 16:19:16 -0400
Subject: [idle] intel_idle: Fixed C6 state on Avoton/Rangeley processors

Message-id: <20140811161916.48098.89723.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 87634
O-Subject: [PATCH RHEL7.1 BZ1048050 2/3] intel_idle: Fixed C6 state on Avoton/Rangeley processors
Bugzilla: 1048050
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1048050

Description:
------------
Corrected the MWAIT flag for C-State C6 on Intel Avoton/Rangeley processors.

Signed-off-by: Arne Bockholdt <linux-kernel@bockholdt.com>
Acked-by: Len Brown <len.brown@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

RHEL Version Found:
-------------------
7.0

Upstream Status:
----------------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=22e580d07f6529a395c129575127ea6d860aed3a

----------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/idle/intel_idle.c b/drivers/idle/intel_idle.c
index 1d8d35e..d168122 100644
--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -428,7 +428,7 @@ static struct cpuidle_state atom_cstates[] = {
  {
   .enter = NULL }
 };
-static struct cpuidle_state avn_cstates[CPUIDLE_STATE_MAX] = {
+static struct cpuidle_state avn_cstates[] __initdata = {
  {
   .name = "C1-AVN",
   .desc = "MWAIT 0x00",
@@ -439,7 +439,7 @@ static struct cpuidle_state avn_cstates[CPUIDLE_STATE_MAX] = {
  {
   .name = "C6-AVN",
   .desc = "MWAIT 0x51",
-  .flags = MWAIT2flg(0x58) | CPUIDLE_FLAG_TIME_VALID | CPUIDLE_FLAG_TLB_FLUSHED,
+  .flags = MWAIT2flg(0x51) | CPUIDLE_FLAG_TIME_VALID | CPUIDLE_FLAG_TLB_FLUSHED,
   .exit_latency = 15,
   .target_residency = 45,
   .enter = &intel_idle },
-- 
1.7.1